addons/binding/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/internal/lib/device/connection/WifiConnection.java

```diff
@@ -68,39 +67,36 @@ public boolean connect() {
             logger.debug("{}: socket not null, return!", TAG);
             return true;
         }
-        mConnectThread = new Thread(new Runnable() {
-            @Override
-            public void run() {
-                try {
-                    mCmdRun = true;
-                    logger.debug("{}: connect device! {}, {}", TAG, mDevice.getAddress(), mDevice.getPort());
-                    mSocket = new Socket(mDevice.getAddress(), mDevice.getPort());
-                    mSocket.setKeepAlive(true);
-                    mWriter = new BufferedOutputStream(mSocket.getOutputStream());
-                    mReader = new BufferedReader(new InputStreamReader(mSocket.getInputStream()));
-                    mDevice.setConnectionState(ConnectState.CONNECTED);
-                    while (mCmdRun) {
-                        try {
-                            String value = mReader.readLine();
-                            logger.debug("{}: get response: {}", TAG, value);
-                            if (value == null) {
-                                mCmdRun = false;
-                            } else {
-                                mDevice.onNotify(value);
-                            }
-                        } catch (Exception e) {
-                            logger.debug("Exception: {}", e);
+        mConnectThread = new Thread(() -> {
+            try {
+                mCmdRun = true;
+                logger.debug("{}: connect device! {}, {}", TAG, mDevice.getAddress(), mDevice.getPort());
+                mSocket = new Socket(mDevice.getAddress(), mDevice.getPort());
+                mSocket.setKeepAlive(true);
+                mWriter = new BufferedOutputStream(mSocket.getOutputStream());
```

Comment:
I am not sure but shouldn't we maybe use a `try-with-resources` statement here (and the following line) to be on the safe side?

Reply:
Even worse. Than we have to make sure it / they (`mWriter` and `mReader`) got closed in the `disconnect` method?

Reply:
I don't see any other method calling the `disconnect` method. Maybe @claell knows how this should work?

Reply:
I know that I did not really understand the logic of this code the last time I got in touch with it. So I probably cannot be of help here unfortunately.
What I do know however is that it probably has some flaws in it, as it seems to have problems to catch up again after a device lost its connection during runtime (like turning it off and on again). Maybe this is related to problems with properly disconnecting a device?

Reply:
Alright. For now I would say we leave this PR like it is and keep in mind that the connection handling could be improved some time in the future.

Reply:
@wborn feel free to merge if you agree.

Reply:
Yes the code can use some further improvements and fixes that can be done in follow up PRs.